ci/dev-env-push.py

```diff
@@ -104,7 +104,7 @@ def main():
         shutil.rmtree(nix_cache_dir)
 
     # copy to nix cache
-    cmd = ["nix", "copy", "--to", store_url, "-f", "./nix", "tools", "ci-cached"]
+    cmd = ["nix", "--extra-experimental-features", "nix-command", "copy", "--to", store_url, "-f", "./nix", "tools", "ci-cached"]
```

Comment:
I’ve gone for the CLI flag since that doesn’t result in a warning on nix 2.3 for people locally since this script is only run by CI and afaict this is our only usage of the nix command (which probably could be replaced but then again it worked fine forever so no real reason to).

Reply:
apparently on macos we still have an older version of nix because our setup is a mess so went back to the config file. While it produces a warning it works across both versions at least.